lib/web/scripts/node-join/install.sh

```diff
@@ -758,87 +760,177 @@ if teleport_binaries_exist; then
+            PACKAGE_MANAGER_COMMAND="dnf -y install"
+        elif check_exists yum; then
+            log "Found 'yum' package manager, using it"
+            PACKAGE_MANAGER_COMMAND="yum -y localinstall"
+        else
+            PACKAGE_MANAGER_COMMAND=""
+            log "Cannot find 'yum' or 'dnf' package manager commands, will try installing the rpm manually instead"
+        fi
+        # check that needed tools are installed
+        check_exists_fatal curl
+        log "Downloading Teleport ${TELEPORT_FORMAT} release ${TELEPORT_VERSION}"
+        DOWNLOAD_FILENAME=$(get_download_filename "${URL}")
+        download "${URL}" "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
+        # install with package manager if available
+        if [[ ${PACKAGE_MANAGER_COMMAND} != "" ]]; then
+            log "Installing Teleport release from ${TEMP_DIR}/${DOWNLOAD_FILENAME} using ${PACKAGE_MANAGER_COMMAND}"
+            # install rpm with package manager
+            ${PACKAGE_MANAGER_COMMAND} "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
+        # use rpm if we couldn't find a package manager
+        else
+            # install RPM (in upgrade mode)
+            log "Using rpm to install ${TEMP_DIR}/${DOWNLOAD_FILENAME}"
+            rpm -Uvh "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
+        fi
     else
-        PACKAGE_MANAGER_COMMAND=""
-        log "Cannot find 'yum' or 'dnf' package manager commands, will try installing the rpm manually instead"
-    fi
-    # check that needed tools are installed
-    check_exists_fatal curl
-    log "Downloading Teleport ${TELEPORT_FORMAT} release ${TELEPORT_VERSION}"
-    DOWNLOAD_FILENAME=$(get_download_filename "${URL}")
-    download "${URL}" "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
-    # install with package manager if available
-    if [[ ${PACKAGE_MANAGER_COMMAND} != "" ]]; then
-        log "Installing Teleport release from ${TEMP_DIR}/${DOWNLOAD_FILENAME} using ${PACKAGE_MANAGER_COMMAND}"
-        # install rpm with package manager
-        ${PACKAGE_MANAGER_COMMAND} "${TEMP_DIR}/${DOWNLOAD_FILENAME}"
-    # use rpm if we couldn't find a package manager
+        log_important "Can't figure out what Teleport format to use"
+        exit 1
+    fi
+}
+
+install_from_repo(){
+    if [[ "${REPO_CHANNEL}" == "" ]]; then
+        # By default, use the current version's channel.
+        REPO_CHANNEL=v"${TELEPORT_VERSION//.*/}"
+    fi
+
+    # Populate $ID, $VERSION_ID, $VERSION_CODENAME and other env vars identifying the OS.
+    # shellcheck disable=SC1091
+    . /etc/os-release
+
+    if [ "$ID" == "debian" ] || [ "$ID" == "ubuntu" ]; then
+        # old versions of ubuntu require that keys get added by `apt-key add`, without
+        # adding the key apt shows a key signing error when installing teleport.
+        if [[
+            ($ID == "ubuntu" && $VERSION_ID == "16.04") || \
+            ($ID == "debian" && $VERSION_ID == "9" )
+        ]]; then
+            sudo apt install apt-transport-https gnupg -y
```

Comment:
it should not be necessary to use `sudo` since the script ensure it runs as root (`assert_running_as_root`). Same comment for the other places `sudo` is used.

Reply:
Indeed, thank you :+1: